src/lib.rs

```diff
@@ -624,15 +389,25 @@ impl AstarteSdk {
     where
         D: Into<AstarteType>,
     {
-        trace!("sending {} {}", interface_name, interface_path);
+        debug!("sending {} {}", interface_name, interface_path);
 
-        let buf = AstarteSdk::serialize_individual(data, timestamp)?;
+        let data: AstarteType = data.into();
+
+        let buf = AstarteSdk::serialize_individual(data.clone(), timestamp)?;
 
         if cfg!(debug_assertions) {
             self.interfaces
                 .validate_send(interface_name, interface_path, &buf, &timestamp)?;
         }
 
+        if self
+            .check_property_on_send(interface_name, interface_path, data.clone())
+            .await?
+        {
+            debug!("property was already sent, no need to send it again");
+            return Ok(());
+        }
+
```

Comment:
There should be this kind of check also for unset (to avoid resending the unset)